sq/src/commands/sign.rs

```diff
@@ -19,31 +19,34 @@ use crate::openpgp::serialize::stream::{
     Message, Armorer, Signer, LiteralWriter,
 };
 use crate::openpgp::types::SignatureType;
-use crate::{
-    Config,
-};
+
+use crate::secrets::PreSecret;
+
+use crate::Config;
 
 pub fn sign(config: Config,
             input: &mut (dyn io::Read + Sync + Send),
             output_path: Option<&str>,
-            secrets: Vec<openpgp::Cert>, detached: bool, binary: bool,
+            secrets: Vec<PreSecret>,
+            detached: bool, binary: bool,
             append: bool, notarize: bool, time: Option<SystemTime>,
             notations: &[(bool, NotationData)])
             -> Result<()> {
     match (detached, append|notarize) {
         (_, false) | (true, true) =>
             sign_data(config, input, output_path, secrets, detached, binary,
-                      append, time, notations),
+                append, time, notations),
         (false, true) =>
             sign_message(config, input, output_path, secrets, binary, notarize,
-                         time, notations),
+                time, notations),
     }
 }
 
 fn sign_data(config: Config,
              input: &mut dyn io::Read, output_path: Option<&str>,
-             secrets: Vec<openpgp::Cert>, detached: bool, binary: bool,
-             append: bool, time: Option<SystemTime>,
+             presecrets: Vec<PreSecret>,
+             detached: bool, binary: bool, append: bool,
+             time: Option<SystemTime>,
```

Comment:
Nitpick: you may want to split these parameters over multiple lines. Now it's a bit hard to read